lib/whitehall_importer/integrity_checker.rb

```diff
@@ -136,5 +191,43 @@ def default_image?(proposed_image_payload, publishing_api_image, attribute)
         proposed_image_payload.empty? &&
         publishing_api_image[attribute] == "placeholder"
     end
+
+    def withdrawal_type_matches?
+      publishing_api_content.dig(:unpublishing, :type) == "withdrawal" &&
+        edition.state == "withdrawn"
+    end
+
+    def removed_type_matches?
+      publishing_api_type = publishing_api_content.dig(:unpublishing, :type)
+      %w[gone redirect].include?(publishing_api_type) && edition.state == "removed"
+    end
+
+    def unpublishing_time_matches?
+      publishing_api_unpublished_at = publishing_api_content.dig(:unpublishing,
+                                                                 :unpublished_at)
+      Time.zone.iso8601(publishing_api_unpublished_at) ==
+        edition.status.details.withdrawn_at
```

Comment:
So I've found an issue with converting the Publishing API timestamp to rfc3339 here because the format seems to be different for `unpublished_at` compared with other timestamps like `public_updated_at`.  After testing on integration, `unpublished_at` is coming back in this format: `"2020-03-02T11:26:52"` whereas `public_updated_at` is coming back like this: `"2020-03-11T12:00:00Z"`.  The consequence of this is when trying to parse it e.g. `Time.zone.rfc3339("2020-03-02T11:26:52")` we get an argument error (this was tried on integration).  I'm going leave it to parse to `iso8601` for now and we may need to rethink the comparison method a bit.

Reply:
oh shoot - it's coming back with no timezone information? that might mean we'll get the time the time wrong for the half year in BST. I'm just going to have a look at Pub API for this.

Reply:
I've just opened this https://github.com/alphagov/publishing-api/pull/1746

Reply:
I've now changed this to convert to `rfc3339`.  It's working as expected on integration.